File: assistant/ai-assistant-api-unused/routers/communities.ts
import { Request, Response } from 'express';
import { upsertCommunityInVectorStore, upsertPostInVectorStore } from './engine/vector_store.js';
import { Community } from './models/community.js';
import { Post } from './models/post.js';
import weaviate from 'weaviate';
import * as json from 'json';

const communityRouter = express.Router();
const client = new weaviate.Client("http://localhost:8080");

function getCommunityFromStore(clusterId: number, communityId: number, attributes: string[] = [
    "communityId", "name", "assistantConfiguration", "cluster_id"
]): any {
    const whereFilter = {
        "operator": "And",
        "operands": [
            {
                "path": ["cluster_id"],
                "operator": "Equal",
                "valueInt": clusterId
            },
            {
                "path": ["communityId"],
                "operator": "Equal",
                "valueInt": communityId
            }
        ]
    };

    return client.query
        .get("Communities", attributes)
        .withLimit(1)
        .withWhere(whereFilter)
        .do();
}

communityRouter.put("/api/v1/communities/:clusterId/:communityId", async (req: Request, res: Response) => {
    const clusterId = parseInt(req.params.clusterId);
    const communityId = parseInt(req.params.communityId);
    const community: Community = req.body;
    let communityFound = false;

    try {
        const result = getCommunityFromStore(clusterId, communityId, ["communityId", "cluster_id"]);
        console.log(`result: ${result}`);
        if (result["data"]["Get"]["Communities"]) {
            communityFound = true;
        } else {
            communityFound = false;
        }
    } catch {
        communityFound = false;
    }

    if (!communityFound) {
        community.communityId = communityId;
        community.clusterId = clusterId;
        await upsertCommunityInVectorStore(community);
    }

    res.status(200).send("OK");
});

communityRouter.get("/api/v1/communities/:clusterId/:communityId/:language", async (req: Request, res: Response) => {
    const clusterId = parseInt(req.params.clusterId);
    const communityId = parseInt(req.params.communityId);
    const language = req.params.language;
    const result = getCommunityFromStore(clusterId, communityId);
    console.log(language);

    const assistantConfiguration = json.loads(result["data"]["Get"]["Communities"][0]["assistantConfiguration"]);
    const welcomeMessages = assistantConfiguration["welcomeMessage"];
    const textInputLabels = assistantConfiguration["textInputLabel"];
    const themeMainColor = assistantConfiguration["theme"]["mainColor"];

    console.log(welcomeMessages["en"]);

    let welcomeMessage;
    if (welcomeMessages[language]) {
        welcomeMessage = welcomeMessages[language];
    } else {
        welcomeMessage = welcomeMessages["en"];
    }

    let textInputLabel;
    if (textInputLabels[language]) {
        textInputLabel = textInputLabels[language];
    } else {
        textInputLabel = textInputLabels["en"];
    }

    result["data"]["Get"]["Communities"][0]["welcomeMessage"] = welcomeMessage;
    result["data"]["Get"]["Communities"][0]["textInputLabel"] = textInputLabel;
    result["data"]["Get"]["Communities"][0]["themeMainColor"] = themeMainColor;
    result["data"]["Get"]["Communities"][0]["assistantConfiguration"] = null;

    const jsonStr = JSON.stringify(result["data"]["Get"]["Communities"][0], null, 4);
    console.log(jsonStr);
    res.status(200).send(jsonStr);
});

export default communityRouter;

File: assistant/ai-assistant-api-old-python/routers/communities.py
from requests import Request
from engine.vector_store import upsert_community_in_vector_store, upsert_post_in_vector_store
from fastapi import APIRouter, Response
from models.community import Community
from models.post import Post
import weaviate
import json

community_router = APIRouter()
client = weaviate.Client("http://localhost:8080")

def get_community_from_store(cluster_id, community_id, attributes=[
    "communityId", "name", "assistantConfiguration", "cluster_id"
    ]):

    where_filter = {
        "operator": "And",
        "operands": [
            {
                "path": ["cluster_id"],
                "operator": "Equal",
                "valueInt": cluster_id
            },
            {
                "path": ["communityId"],
                "operator": "Equal",
                "valueInt": community_id
            }
        ]
    }

    #print(f"where_filter: {where_filter}")

    return (
        client.query.
        get("Communities", attributes).
        with_limit(1).
        with_where(where_filter).
        do()
    )


@community_router.put("/api/v1/communities/{cluster_id}/{community_id}")
async def update_community(cluster_id: int, community_id: int, community: Community):
    communityFound = False

    try:
        result = get_community_from_store(cluster_id, community_id, attributes=["communityId","cluster_id"])
        print(f"result: {result}")
        if result["data"]["Get"]["Communities"]:
            communityFound = True
        else:
            communityFound = False
    except:
        communityFound = False

    if communityFound == False:
        community.community_id = community_id
        community.cluster_id = cluster_id
        await upsert_community_in_vector_store(community)

    return Response(content="OK", status_code=200)


@community_router.get("/api/v1/communities/{cluster_id}/{community_id}/{language}")
async def get_community(cluster_id: int, community_id: int, language: str):
    result = get_community_from_store(cluster_id, community_id)
    print(language)

    #print(result)
    #print(result["data"]["Get"]["Communities"][0])
    #print(result["data"]["Get"]["Communities"][0]["assistantConfiguration"])
    assistant_configuration = json.loads(result["data"]["Get"]["Communities"][0]["assistantConfiguration"])
    #print(assistant_configuration)
    welcome_messages = assistant_configuration["welcomeMessage"]
    text_input_labels = assistant_configuration["textInputLabel"]
    theme_main_color = assistant_configuration["theme"]["mainColor"]

    print(welcome_messages["en"])

    if welcome_messages[language]:
        welcome_message = welcome_messages[language]
    else:
        welcome_message =  welcome_messages["en"]

    if text_input_labels[language]:
          text_input_label = text_input_labels[language]
    else:
        text_input_label =  text_input_labels["en"]

    result["data"]["Get"]["Communities"][0]["welcomeMessage"] = welcome_message
    result["data"]["Get"]["Communities"][0]["textInputLabel"] = text_input_label
    result["data"]["Get"]["Communities"][0]["themeMainColor"] = theme_main_color
    result["data"]["Get"]["Communities"][0]["assistantConfiguration"] = None

    # result["data"]["Get"][self._index_name]:
    json_str = json.dumps(result["data"]["Get"]
                          ["Communities"][0], indent=4, default=str)
    print(json_str)
    return Response(content=json_str, status_code=200)
